src/Workspaces/MSBuild/Core/MSBuild/MSBuildProjectLoader.cs

```diff
@@ -57,12 +58,20 @@ internal MSBuildProjectLoader(
     /// <param name="properties">An optional dictionary of additional MSBuild properties and values to use when loading projects.
     /// These are the same properties that are passed to MSBuild via the /property:&lt;n&gt;=&lt;v&gt; command line argument.</param>
     public MSBuildProjectLoader(Workspace workspace, ImmutableDictionary<string, string>? properties = null)
+        : this(workspace, properties, projectFileInfoLoaderFactory: null)
+    {
+    }
+
+    internal MSBuildProjectLoader(
+        Workspace workspace,
```

Comment:
Seems like the only two things that are needed from workspace are `SolutionServices` and `OnWorkspaceFailed`. Can we remove the dependency on Workspace from DiagnosticReporter and here?